### PR TITLE
Throwing knife bandolier

### PR DESCRIPTION
**Pull request by ashGlaw**
#### Summary

```SUMMARY: Content "Adds a bandolier for throwing knives"```
#### Purpose of change
Other weapons and ammo can be stored in wearable containers, and I wanted throwing knives to get some love. 

#### Describe the solution
I added the item to bandoliers.json, and I added the "SHEATH_KNIFE" flag to throwing_knife in archery.json. I made the interaction speed of  the bandolier pretty low, but not the lowest of the sheaths/holsters. It should be lower than getting the knives out of inventory anyway.

I also fixed a syntax error in med.json that was keeping the game from loading.

#### Describe alternatives you've considered
I'm thinking about also making a small wrist or ankle sheath for throwing knives that takes 2 blades. It would be swell if, on activation of the bandolier, one of the available option was to fully load the bandolier all at once.

The holster UI for multi holsters is pretty cumbersome. I worry that people won't bother using the bandolier because it takes more effort

#### Testing
I spawned the bandolier in the game, loaded it up with throwing knives, drew one and threw it. All worked fine.

#### Additional context
Currently, I'm not sure if throwing knives are the only knives that can be stored in the bandolier. I'm also not sure if that's an issue. Putting a special flag on throwing knives that's only called by the bandolier could be a solution, but that feels inelegant to me. What do people think?

If people dig it I could make a crafting recipe and add it to lists

### CODE REVIEW

**Comment by Brian-Otten** [comment]
Did you mean to slightly edit melatonin in the same PR?

Throwing knife bandolier looks good to me, doesn't sound like a problem if it's also able to hold other knives that have the exact same volume as the throwing knives.

Being able to automatically fully load it might be nice, but would handle differently from other sheaths/pouches, so this probably shouldn't be a special exception. Could someday be a nice feature of such items.

**Comment by TechyBen** [comment]
Best to keep other bugs/edits in a separate PR AFAIK.

File: data/json/items/armor/bandolier.json [hunk 1]
```diff
@@ -263,5 +263,40 @@
...
     },
     "flags": [ "WATER_FRIENDLY", "BELTED" ]
-  }
+  },
+  {
+  "id": "throwingknifebandolier",
+  "type": "ARMOR",
+  "name": "throwing knife bandolier",
+  "name_plural": "throwing knife bandoliers",
+  "description": "A leather bandolier strapped around the torso for keeping throwing knives close to hand..",
```
Comment (Rail-Runner) [inline data/json/items/armor/bandolier.json]:
```suggestion
  "description": "A leather bandolier strapped around the torso for keeping throwing knives close to hand.",
```

**Comment by RarkGrames** [comment]
You need to add a recipe or spawn points/item group entries (preferably both). Otherwise it becomes dead code not used anywhere.

**Comment by ashGlaw** [comment]
> 
> 
> You need to add a recipe or item group entries (preferably both). Otherwise it becomes dead code not used anywhere.

Recipe's easy. Item groups are something I haven't touched before. I'm not sure where the best place to put it would be. Any advice on where the bandolier belongs?

**Comment by ashGlaw** [comment]
> Did you mean to slightly edit melatonin in the same PR?
Yes. The duration was written in a "16 h" when it needed to be an integer. I fixed it so I could test. I can take it out later on
> Throwing knife bandolier looks good to me, doesn't sound like a problem if it's also able to hold other knives that have the exact same volume as the throwing knives.
Great. Thanks! 
> Being able to automatically fully load it might be nice, but would handle differently from other sheaths/pouches, so this probably shouldn't be a special exception. Could someday be a nice feature of such items.
Gree. No need to make this bandolier special, but maybe any holster with multi deserves the option to fully load it with one type of item later on. That's a problem for the future, though.

**Comment by anothersimulacrum** [comment]
> Yes. The duration was written in as "16 h" when it needed to be an integer. I fixed it so I could test. I can take it out later on

You should update the binary you're using for testing, when I changed it to that I added support for `time_duration` strings.

**Comment by RarkGrames** [comment]
> 
> Recipe's easy. Item groups are something I haven't touched before. I'm not sure where the best place to put it would be. Any advice on where the bandolier belongs?

I couldn't think of any shop or similar place where it would be reasonable to spawn, but survivor zombies could drop it.

If that is fine for you, you can add an entry in the `survivorzed_extra `item group in \itemgroups\clothing.json, with a spawn weight of 5.

**Comment by Brian-Otten** [comment]
Survivor zed would be a pretty good place for this to spawn, with a low probability.

**Comment by ashGlaw** [comment]
I’ve expanded the pr to include a double throwing knife shesth for the ankle, and a single knife sheath for the wrist, recipes for both, and I added them to the list RarkGrames and Brian-Otten suggested. 

The items are tested. I’ll leave testing the remaining two recipes and making sure survivor zombies drop them for the next time I’m at the computer

**Comment by Shodan14** [comment]
I think we need more multi-purpose bandoliers at this point.

**Comment by ashGlaw** [comment]
How do you mean?

**Comment by Shodan14** [comment]
> 
> 
> How do you mean?
Like one bandolier that can hold any type of ammo and one that can hold any throwing items from rocks to knives to grenades.

**Comment by ashGlaw** [comment]
Well, I think one issue is that different ammo/thrown weapons have different volumes, so the amount you can fit into the bandolier would be different for each type of ammo, wouldn't it?

**Comment by Shodan14** [comment]
> 
> 
> Well, I think one issue is that different ammo/thrown weapons have different volumes, so the amount you can fit into the bandolier would be different for each type of ammo, wouldn't it?
> 
> Maybe general bandoliers would work well for the simplified guns mod

If you imagine it just having elastic straps, MOLLE-style you could attach a number of different things.

**Comment by ZhilkinSerg** [comment]
Feel free to reopen if you want working on it again.

**Pull request closed without merging**